core/src/stages/generate_random_pose.cpp

```diff
@@ -0,0 +1,162 @@
+	auto& p = properties();
+	p.declare<geometry_msgs::PoseStamped>("pose", "target pose to pass on in spawned states");
+	p.declare<size_t>("max_solutions", 20,
+	                  "limit of the number of spawned solution in case randomized sampling is enabled");
+	p.property("timeout").setDefaultValue(1.0 /* seconds */);
+}
+
+template <>
+GenerateRandomPose::PoseDimensionSampler
+GenerateRandomPose::getPoseDimensionSampler<std::normal_distribution>(double stddev) {
+	return [ stddev, &gen = gen_ ](double mean) {
+		static std::normal_distribution<double> dist(mean, stddev);
+		return dist(gen);
+	};
+}
+
+template <>
+GenerateRandomPose::PoseDimensionSampler
+GenerateRandomPose::getPoseDimensionSampler<std::uniform_real_distribution>(double range) {
+	return [ range, &gen = gen_ ](double mean) {
+		static std::uniform_real_distribution<double> dist(mean - 0.5 * range, mean + 0.5 * range);
+		return dist(gen);
+	};
+}
+
+bool GenerateRandomPose::canCompute() const {
+	return GeneratePose::canCompute() && !pose_dimension_samplers_.empty();
+}
+
+void GenerateRandomPose::compute() {
+	if (upstream_solutions_.empty())
+		return;
+
+	planning_scene::PlanningScenePtr scene = upstream_solutions_.pop()->end()->scene()->diff();
+	geometry_msgs::PoseStamped target_pose = properties().get<geometry_msgs::PoseStamped>("pose");
+	if (target_pose.header.frame_id.empty())
+		target_pose.header.frame_id = scene->getPlanningFrame();
+	else if (!scene->knowsFrameTransform(target_pose.header.frame_id)) {
+		ROS_WARN_NAMED("GenerateRandomPose", "Unknown frame: '%s'", target_pose.header.frame_id.c_str());
+		return;
+	}
+
+	const auto& spawn_target_pose = [&](const geometry_msgs::PoseStamped& target_pose) {
+		InterfaceState state(scene);
+		state.properties().set("target_pose", target_pose);
+
+		SubTrajectory trajectory;
+		trajectory.setCost(0.0);
+
+		rviz_marker_tools::appendFrame(trajectory.markers(), target_pose, 0.1, "pose frame");
+
+		spawn(std::move(state), std::move(trajectory));
+	};
+
+	// Spawn initial pose
+	spawn_target_pose(target_pose);
+
+	if (pose_dimension_samplers_.empty())
+		return;
+
+	// Use target pose as seed pose
+	geometry_msgs::PoseStamped seed_pose = target_pose;
+	Eigen::Isometry3d seed, sample;
+	tf::poseMsgToEigen(seed_pose.pose, seed);
+	double elapsed_time = 0.0;
+	const auto start_time = std::chrono::steady_clock::now();
+	size_t spawned_solutions = 0;
+	const size_t max_solutions = properties().get<size_t>("max_solutions");
+	while (elapsed_time < timeout() && ++spawned_solutions < max_solutions) {
+		// Randomize pose using specified dimension samplers applied in the order
+		// in which they have been specified
+		sample = seed;
+		for (const auto& pose_dim_sampler : pose_dimension_samplers_) {
+			switch (pose_dim_sampler.first) {
+				case X:
+					sample.translate(Eigen::Vector3d(pose_dim_sampler.second(seed_pose.pose.position.x), 0, 0));
```

Comment:
Looks like it, but a short explanation wouldn't hurt.
So, `sample` is already set to the target pose and translating it with a distance of the target pose dimensions is redundant. I think, the first version, sample was set to identity instead of the target pose.

Reply:
Sorry for not adding an explanation, as you said sample is already set to the target pose so this will add the specified dimension back + random value